src/components/Identity/index.js

```diff
@@ -25,15 +35,54 @@ const addIdsContext = (payload, ecid) => {
 const createIdentity = ({ config, logger, cookie }) => {
   // We avoid reading the ECID from the cookie right away, because we
   // need to wait for the user to opt in first.
-  const getEcid = () => cookie.get(ECID_NAMESPACE);
+  const getEcid = () => cookie.get(EXPERIENCE_CLOUD_ID);
   let optIn;
   let responseRequested = false;
   let deferredForEcid;
+  let network;
+  let lifecycle;
+  const customerIDs = {};
+  const getCustomerIDs = () => customerIDs;
+
+  const makeServerCall = event => {
+    const payload = network.createPayload();
+    payload.addEvent(event);
+    const responsePromise = Promise.resolve()
+      .then(() => {
+        return lifecycle.onBeforeDataCollection(payload, responsePromise);
+      })
+      .then(() => {
+        return network.sendRequest(payload, payload.expectsResponse);
+      });
+
+    return responsePromise;
+  };
+
+  const setCustomerIDs = options => {
+    const event = createEvent();
+    validateCustomerIDs(options.customerIDs);
+    Object.keys(options.customerIDs).forEach(key => {
+      customerIDs[key] = options.customerIDs[key];
+    });
+    const normalizedCustomerIDs = normalizeCustomerIDs(customerIDs);
+    const customerIDsHash = createHashFromString(
+      serializeCustomerIDs(normalizedCustomerIDs)
+    );
+    const hasSynced = customerIDsHash === cookie.get(CUSTOMER_ID_HASH);
+    event.mergeMeta({ identity: { customerIDs, hasSynced } });
```

Comment:
Is that how Konductor expects those? Just double checking that we discussed them with Rares' team.

Reply:
I will double check with Rares in the morning.

Reply:
@jileeshs If I read this correctly, it looks like you're passing both the (declared) customer IDs and a flag to specify that _nothing changed_ in the same object, under `meta.identity`.

_We need to tweak this, as Konductor expects identities to be sent in the `identityMap`._ Identities can be either set at payloads' root level, or at event level. 

What's interesting to notice further is that this part of context data is deeply merged in Konductor: event-level `identityMap` gets merged with the root level `identityMap` entries; this helps you, for example, set the ECID at root level and then have customerIDs sent in at event level, when they become available to Alloy. This is particularly handy when you pass in a batch of events to Konductor -- if only one event is sent, it does not matter where you declare the identities (root or event level).

To sum up:
1. Declared customer IDs need to be sent in `identityMap`, either as root-level or event-level context data
2. To limit the number of calls we send in to the `IdentityService` downstream of Konductor, we need that extra information in `metadata`. It can be either a flag which says "noting changed" (as it looks like it is now) or the opposite, "a new identity has been declared".

Reply:
Thank you @raresvla.

Should we go with:

```
metadata: {
    identity: { // Do we need this?
        newDeclaredId: true|false
    }
}
```
@Aaronius @jileeshs

Reply:
@jileeshs Once we get an answer on this, let's update the PR, close this one and re-open a new one that includes all the new changes from yesterday. Thanks!

Reply:
Maybe `customerIdAdded` or `customerIdChanged` (whichever is more technically accurate) instead of `newDeclaredId`? I feel like `customerIdAdded` might suggest that the value should be a boolean more than `newDeclaredId`. The term "declared" seems new too. If it's used elsewhere or we feel like it helps specify what we're talking about, then that might be good to use, but otherwise I would say we stick to the terms we already have (e.g., "customer ID").

Reply:
@jileeshs I agree with @Aaronius, let's go with `customerIdChanged` since it's more accurate because an ID is really the ID itself and the Auth State; so the ID can stay the same but the auth state can change.

Reply:
Thank you @Aaronius!

Reply:
The ship has probably sailed on this one, but why "customer" and not "user"? The term "customer" seems to suggest an ecommerce context, which doesn't work well with free providers (publishers, social networks, etc). If "customer" is too engrained at this point, just let me know and I'll go on my way...

Reply:
I always thought of it as in Adobe's Customer's Custom ID.  i.e. the ID that is owned by Adobe's Customer.

Reply:
Oh that's interesting. I never thought of it that way.